pkg/member/member_control.go

```diff
@@ -258,11 +258,17 @@ func (m *memberControl) doPromoteMember(ctx context.Context, member *etcdserverp
 		m.logger.Info("Member promoted ", member.Name, " : ", member.ID)
 		return nil
 	} else if errors.Is(err, rpctypes.Error(rpctypes.ErrGRPCMemberNotLearner)) { //Member is not a learner
-		if member.PeerURLs[0] == "http://localhost:2380" { //[]string{"http://localhost:2380"}[0] {
+		if member.PeerURLs[0] == "http://localhost:2380" {
 			// Already existing clusters have `http://localhost:2380` as the peer address. This needs to explicitly updated to the new address
 			// TODO: Remove this peer address updation logic on etcd-br v0.20.0
-			err = m.updateMemberPeerAddress(ctx, member.ID)
-			m.logger.Errorf("Could not update member peer URL : %v", err)
+			err := retry.OnError(retry.DefaultBackoff, func(err error) bool {
+				return err != nil
+			}, func() error {
+				return m.updateMemberPeerAddress(ctx, member.ID)
+			})
+			if err != nil {
+				m.logger.Warnf("Could not update member peer URL : %v", err)
```

Comment:
Instead of warning ... can you log the error.

Reply:
Done